### DISCUSSION TITLE
Best way to handle auto-updates

### DISCUSSION THREAD

**Discussion by camimaya21** [SkillOnBoard/SnippBoard#53] [Polls]
I have been searching about different ways (without investing in fancy infra 💸 ) AKA "Free" ways to have auto-updates for the Electron App.

Option A:
The best option that I have found so far is to use [Electron Build](https://github.com/electron-userland/electron-builder) https://www.electron.build/auto-update.html which allow us:

PROS:
- to connect a private repo to get the updates
- Don't need a dedicated server to handle the updates
- Looks like pretty simple to configure
- Pretty much "free"

CONS:
- I think is only worth to be used for the MVP, I'm not quite happy about to generate a GH Token Key to configure the updates.
- This does not scalate well due to Github API limit request (The GitHub API currently has a rate limit of 5000 requests per user per hour. An update check uses up to 3 requests per check.) https://www.electron.build/auto-update.html#private-github-update-repo

Option B:

Using Electron Builder as well to avoid having a dedicated server for the updates, but perhaps we might need to hired S3 or Digital ocean for storing the artifacts instead of using the GH releases which requires a token to work.

Option C:

Another option, something that I have seen in [Reddit](https://www.reddit.com/r/electronjs/comments/1f5a35g/auto_update_private_app/) and TBH sounds worth to try 😏, we could create a Public repository inside of the organisation where we could publish only the builds (binary) as code base and keep the same releases and tags like we have in the private repo. By doing this:

- We can take advantage to have also the Public pages, and host in Github our landing page and also we can use directly the connection between Electron Build and Github Releases, avoiding exposing an Github Token.
- This might require to have a proper CI pipeline or to update the release script to publish in both repos the same versions/releases/tags, I prefer the CI to avoid scalability problems.

**Comment by camimaya21** [comment]
Ahhh and I almost forgot, all of the updates only work if the app is properly signed for Apple, so back to square one 😢 

![shit](https://i.giphy.com/media/v1.Y2lkPTc5MGI3NjExdWcydHFwemx5YXQ5d2I5NWZ6c2VreDk4dGZwbzdhOXI4emd0MmJpaiZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/l2JdX1lCEu2NKlNf2/giphy.gif)

**Comment by camimaya21** [comment]
For this MVP and until we decide what to do next, I will vote for the Option A. Then perhaps migrate to Option C and then a proper Option B with a proper CI pipeline and a 3rd party who should host the artifacts + landing, lika a S3 or something like that.

**Comment by RubenMZ** [comment]
Awesome 🎉 🎉 🎉  I would go for the option A but the option C looks interesting for future iterations :)

**Reply by camimaya21** [reply]
We decided to go on Option C but avoid creating the CI pipeline, the new versions will be handle manually in that repo
